### DISCUSSION TITLE
auto-sleep for afk

### DISCUSSION THREAD

**Discussion by VuHoangQuan2008** [PrismarineJS/mineflayer#3106] [Q&A]
I made a bot that can sleep when you type "sleep" in the chat but I had a auto-crop farm so I need the bot always sleep when it is night time. Someone please help me, I can't figure this out. Or any solution that the autofarms always work in daytime?
![image](https://github.com/PrismarineJS/mineflayer/assets/87169586/fb8938bf-d3f5-4bf8-a420-a945482a6e9d)
![image](https://github.com/PrismarineJS/mineflayer/assets/87169586/2cc93f93-523a-4beb-a1c7-24a8303b3978)
Code:
const mineflayer = require('mineflayer')

const bot = mineflayer.createBot({
  host: '192.168.1.xxx',
  username: 'YoBot',
  auth: 'offline' 
})

function createBot () {
    bot.once('spawn', () => {
        bot.chat("/gamemode creative")
    })
    bot.on('chat', (username, message) => {
        if (username === bot.username) return
        switch (message) {
          case 'sleep':
            goToSleep()
            break
          case 'wakeup':
            wakeUp()
            break
        }
      })
      
      bot.on('sleep', () => {
        bot.chat('Good night!')
      })
      bot.on('wake', () => {
        bot.chat('Good morning!')
      })
      
      async function goToSleep () {
        const bed = bot.findBlock({
          matching: block => bot.isABed(block)
        })
        if (bed) {
          try {
            await bot.sleep(bed)
            bot.chat("I'm sleeping")
          } catch (err) {
            bot.chat(`I can't sleep: ${err.message}`)
          }
        } else {
          bot.chat('No nearby bed')
        }
      }
      
      async function wakeUp () {
        try {
          await bot.wake()
        } catch (err) {
          bot.chat(`I can't wake up: ${err.message}`)
        }
      }
    bot.on('error', (err) => console.log(err))
    // Log errors and kick reasons:
    bot.on('kicked', console.log)
    bot.on('error', console.log)
    bot.on('end', createBot)
  }
  

createBot()

**Comment by PondWader** [comment]
You can use the time event (https://github.com/PrismarineJS/mineflayer/blob/master/docs/api.md#time) and then use `bot.time` to figure out if it's night and sleep
